src/index.js

```diff
@@ -60,12 +58,34 @@ const createProxyfied = (state, cache) => {
     trapped = proxyState(state, null, cache && cache.proxy);
     if (cache) cache.trapped.set(state, trapped);
   }
+  return trapped;
+};
+
+// track state usage in selector, and only rerun if necessary
+const runSelector = (state, selector, lastResult) => {
+  if (lastResult) {
+    const {
+      state: lastState,
+      selector: lastSelector,
+      innerTrapped: lastInnerTrapped,
+    } = lastResult;
+    const shouldRerunSelector = selector !== lastSelector || !proxyEqual(
+      lastState,
+      state,
+      lastInnerTrapped.affected,
+    );
+    if (!shouldRerunSelector) {
+      return lastResult;
+    }
+  }
+  const innerTrapped = createTrapped(state);
+  const value = selector(innerTrapped.state);
```

Comment:
`innerTrapped.reset()` before the call?

Reply:
reset() is called in createTrapped.
https://github.com/dai-shi/reactive-react-redux/blob/7affc87c197b65fcd58913c5b9b1063c5baf0653/src/index.js#L56